cypress/e2e/invoices/save-invoise-fiscal-year-after-adding-adjustment.cy.js

```diff
@@ -0,0 +1,295 @@
+
+describe('ui-finance: Fiscal Year Rollover', () => {
+  const firstFiscalYear = { ...FiscalYears.defaultUiFiscalYear };
+  const secondFiscalYear = {
+    name: `autotest_2_year_${getRandomPostfix()}`,
+    code: DateTools.getRandomFiscalYearCode(2000, 9999),
+    periodStart: `${DateTools.getDayTomorrowDateForFiscalYear()}T00:00:00.000+00:00`,
+    periodEnd: `${DateTools.getDayAfterTomorrowDateForFiscalYear()}T00:00:00.000+00:00`,
+    description: `This is fiscal year created by E2E test automation script_${getRandomPostfix()}`,
+    series: 'FY',
+  };
+  const thirdFiscalYear = {
+    name: `autotest_3_year_${getRandomPostfix()}`,
+    code: DateTools.getRandomFiscalYearCode(2000, 9999),
+    periodStart: `${DateTools.getDayAfterTomorrowDateForFiscalYear()}T00:00:00.000+00:00`,
+    periodEnd: `${DateTools.get2DaysAfterTomorrowDateForFiscalYear()}T00:00:00.000+00:00`,
+    description: `This is fiscal year created by E2E test automation script_${getRandomPostfix()}`,
+    series: 'FY',
+  };
+  const defaultLedger = {
+    ...Ledgers.defaultUiLedger,
+    restrictEncumbrance: false,
+    restrictExpenditures: false,
+  };
+  const defaultFund = { ...Funds.defaultUiFund };
+  const defaultOrder = {
+    ...NewOrder.defaultOneTimeOrder,
+    orderType: 'One-time',
+    approved: true,
+    reEncumber: false,
+  };
+  const organization = { ...NewOrganization.defaultUiOrganizations };
+  const invoice = { ...NewInvoice.defaultUiInvoice };
+  const allocatedQuantity = '100';
+  const periodStartForFirstFY = DateTools.getThreePreviousDaysDateForFiscalYearOnUIEdit();
+  const periodEndForFirstFY = DateTools.getPreviousDayDateForFiscalYearOnUIEdit();
+  const periodStartForSecondFY = DateTools.getCurrentDateForFiscalYearOnUIEdit();
+  const periodEndForSecondFY = DateTools.get2DaysAfterTomorrowDateForFiscalYearOnUIEdit();
+  const periodStartForThirdFY = DateTools.getCurrentDateForFiscalYearOnUIEdit();
+  const periodEndForThirdFY = DateTools.get2DaysAfterTomorrowDateForFiscalYearOnUIEdit();
+  const adjustmentDescription = `test_description${getRandomPostfix()}`;
+  const barcode = FinanceHelp.getRandomBarcode();
+  const caption = 'autotestCaption';
+  let user;
+  let orderNumber;
+  let servicePointId;
+  let location;
+
+  before(() => {
+    cy.getAdminToken();
+
+    FiscalYears.createViaApi(firstFiscalYear).then((firstFiscalYearResponse) => {
+      firstFiscalYear.id = firstFiscalYearResponse.id;
+      defaultLedger.fiscalYearOneId = firstFiscalYear.id;
+      Ledgers.createViaApi(defaultLedger).then((ledgerResponse) => {
+        defaultLedger.id = ledgerResponse.id;
+        defaultFund.ledgerId = defaultLedger.id;
+
+        Funds.createViaApi(defaultFund).then((fundResponse) => {
+          defaultFund.id = fundResponse.fund.id;
+
+          cy.loginAsAdmin({ path: TopMenu.fundPath, waiter: Funds.waitLoading });
+          FinanceHelp.searchByName(defaultFund.name);
+          Funds.selectFund(defaultFund.name);
```

Comment:
Please replace UI actions with API calls in preconditions. To see an example please look at 'fiscal-year-is-not-displayed.cy'